drivers/flash/flash_nxp_s32_qspi_nor.c

```diff
@@ -0,0 +1,1106 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT	nxp_s32_qspi_nor
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(nxp_s32_qspi_nor, CONFIG_FLASH_LOG_LEVEL);
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/sys/util.h>
+
+#include <Qspi_Ip.h>
+
+#include "spi_nor.h"
+#include "jesd216.h"
+
+#include "memc_nxp_s32_qspi.h"
+
+#define QSPI_INST_NODE_HAS_PROP_EQ_AND_OR(n, prop, val)				\
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(n, prop),				\
+		(IS_EQ(DT_INST_ENUM_IDX(n, prop), val)),			\
+		(0)) ||
+
+#define QSPI_ANY_INST_HAS_PROP_EQ(prop, val)					\
+	(DT_INST_FOREACH_STATUS_OKAY_VARGS(QSPI_INST_NODE_HAS_PROP_EQ_AND_OR, prop, val) 0)
+
+#define QSPI_INST_NODE_NOT_HAS_PROP_AND_OR(n, prop) \
+	!DT_INST_NODE_HAS_PROP(n, prop) ||
```

Comment:
Not an issue for this PR: maybe it is worth considering proposing `DT_ANY_INST_HAS_PROP_EQ` as general utility macro.